### ISSUE TITLE
Add Keyboard Alternatives To Drag-And-Drop

### ISSUE THREAD

**Issue by MatejGolian** [onionshare/onionshare#1565]
Hello,

For accessibility purposes, it would be nice if there was a keyboard equivalent for dragging-and-dropping files and folders on to Share and Website tabs.

Perhaps the most intuitive way of doing this would be via the CTRL+C and CTRL+V keyboard shortcuts. I guess you already understand how I'd like it to behave, but in case it's clearer, these are the things that should be met in order for this to work:
1. Make sure the OnionShare control for dragging-and-dropping files and folders on to is included in the tab order.
2. Make sure that OnionShare responds to the CTRL+V shortcut when that control has focus.
3. Upon pressing CTRL+V there, make OnionShare analyze the content of the clipboard and in case it contains pointers to files and/or folders, paste it into OnionShare.

Furthermore, I don't know what type of controls are currently used to facilitate drag-and-drop operations in OnionShare, but in case it's needed, it would be great if they could be changed to list views in such manner that it would become possible to interact with the content of those controls at a given time by using standard Windows keyboard shortcuts. Up and down arrows for moving between files, Shift + up/down arrows for making selections, Delete for removing from the list and so on. Alternatively, right-clicking (or pressing the Applications key or Shift+F10) on that list view should popup a context menu with all the available actions (Paste, Remove and so on).
If it's better, I can open a separate issue for this second request. I wanted to point it out here though, because from an end-user perspective these 2 things are closely related.

**Comment by SaptakS** [comment]
I would have thought the "Add files" button would work as a keyboard alternative for drag-and-drop. Is the above mentioned steps the preferred alternative for drag-and-drop?

Also, we don't have a lot of controls after drag-and-drop, but I will test for the things you have mentioned above.

**Comment by SaptakS** [comment]
@MatejGolian Makes sense to me. I will play around with screen readers a little more in different OS. I will give it a shot to implement the copy-paste behaviour that you have said.

**Comment by MatejGolian** [comment]
Thank you.
I don't know if I've mentioned it already, but my primary screen reader is called NVDA. It's completely free and can be set up as a portable program as well, so you can use that for testing too.
Once automatically generated builds per PR become a reality, I'll be more than happy to test things myself.

**Comment by MatejGolian** [comment]
@SaptakS, in case something like you proposed in issue #1580 makes the dialog for adding files more screen reader friendly in the future, I think that there would be strictly no need to imitate drag-and-drop. I've come to this realization after thinking more about this whole thing. I still believe that my original idea is a neat (and especially very quick) way of doing things, but improving the add files dialog for screen reader users would be more than enough.
